sys/auto_init/auto_init.c

```diff
@@ -115,6 +115,10 @@ void auto_init(void)
     DEBUG("Auto init xtimer module.\n");
     xtimer_init();
```

Comment:
Why not instead add a pseudomodule`xtimer_autonit` and have:

`DEFAULT_MODULE+=xtimer_autoinit`

`DISABLE_MODULE+=xtimer_autoinit`

Reply:
The init in the boardfile could be kept.

Reply:
And an `ifdef` in `sys/auto_init/auto_init.c ` around `ximter_init`

Reply:
Well, I'd have to do the same thing with ztimer, and also guard that, and also add another pseudomodule. "ztimer" is a meta module pulling in ztimer_core and the ztimer auto init...

Maybe we should just replace the use of xtimer in nvram_spi. It is only used for "xtimer_spin(_ticks_to_usec(1))", which could be replaced with a simple spin loop.

Or just remove the boot counter. Or does mulle *need* that?

Reply:
> Maybe we should just replace the use of xtimer in nvram_spi. It is only used for "xtimer_spin(_ticks_to_usec(1))", which could be replaced with a simple spin loop.

I think that would be the best fix, it would remove xtimer/ztimer as a dependency.

> Or just remove the boot counter. Or does mulle need that?

I don't have a mulle, but there is one in Berlin.

Reply:
> Well, I'd have to do the same thing with ztimer, and also guard that, and also add another pseudomodule. "ztimer" is a meta module pulling in ztimer_core and the ztimer auto init...

Lets do it with all `auto-init` functions then? I would find it very useful to be able to disable `auto_init` only for specific modules. No extra guard would be needed, only adding `AUTO_INIT` to the existing one.

```
diff --git a/makefiles/defaultmodules.inc.mk b/makefiles/defaultmodules.inc.mk
index 9d8331407c..973983ca8b 100644
--- a/makefiles/defaultmodules.inc.mk
+++ b/makefiles/defaultmodules.inc.mk
@@ -1,3 +1,5 @@
 DEFAULT_MODULE += board cpu core core_msg sys
 
 DEFAULT_MODULE += auto_init
+
+DEFAULT_MODULE += xtimer_auto_init
diff --git a/makefiles/pseudomodules.inc.mk b/makefiles/pseudomodules.inc.mk
index e7f747a3ef..7f3efa5be3 100644
--- a/makefiles/pseudomodules.inc.mk
+++ b/makefiles/pseudomodules.inc.mk
@@ -197,4 +197,7 @@ PSEUDOMODULES += crypto_aes_precalculated
 # This pseudomodule causes a loop in AES to be unrolled (more flash, less CPU)
 PSEUDOMODULES += crypto_aes_unroll
 
+# All auto_init modules are pseudomodules
+PSEUDOMODULES += %_auto_init
+
 # Packages may also add modules to PSEUDOMODULES in their `Makefile.include`.
diff --git a/sys/auto_init/auto_init.c b/sys/auto_init/auto_init.c
index b72b856875..6a69f2a357 100644
--- a/sys/auto_init/auto_init.c
+++ b/sys/auto_init/auto_init.c
@@ -24,7 +24,8 @@
 #include "diskio.h"
 #endif
 
-#ifdef MODULE_XTIMER
+#ifdef MODULE_XTIMER_AUTO_INIT
+#error
 #include "xtimer.h"
 #endif
 
@@ -111,7 +112,7 @@ void auto_init(void)
     void auto_init_random(void);
     auto_init_random();
 #endif
-#ifdef MODULE_XTIMER
+#ifdef MODULE_XTIMER_AUTO_INIT
     DEBUG("Auto init xtimer module.\n");
     xtimer_init();
 #endif
```

That being said removing the `xtimer` dependency is still the better fix IMO. This wold just be a nice feature :)